### PR TITLE
Reland "Shader warm up (#27660)"

### PR DESCRIPTION
**Pull request by liyuqian**
This reverts commit adc8e159a548121a45ff7a45b4e556aff998d4bd.

This should be safe to land once https://github.com/flutter/flutter/pull/28530 gets merged

### CODE REVIEW

File: packages/flutter/lib/src/painting/shader_warm_up.dart [hunk 1]
```diff
@@ -0,0 +1,148 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
```
Comment (jonahwilliams) [inline packages/flutter/lib/src/painting/shader_warm_up.dart]: Should these be 2019?

Comment (liyuqian) [reply packages/flutter/lib/src/painting/shader_warm_up.dart]: Done.

File: examples/layers/test/smoketests/raw/shader_warm_up_test.dart [hunk 2]
```diff
@@ -0,0 +1,13 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
```
Comment (jonahwilliams) [inline examples/layers/test/smoketests/raw/shader_warm_up_test.dart]:
```suggestion
// Copyright 2019 The Chromium Authors. All rights reserved.
```

File: examples/layers/raw/shader_warm_up.dart [hunk 3]
```diff
@@ -0,0 +1,33 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
```
Comment (jonahwilliams) [inline examples/layers/raw/shader_warm_up.dart]:
```suggestion
// Copyright 2019 The Chromium Authors. All rights reserved.
```

File: dev/devicelab/bin/tasks/cubic_bezier_perf__timeline_summary.dart [hunk 4]
```diff
@@ -0,0 +1,14 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
```
Comment (jonahwilliams) [inline dev/devicelab/bin/tasks/cubic_bezier_perf__timeline_summary.dart]:
```suggestion
// Copyright 2019 The Chromium Authors. All rights reserved.
```

File: dev/benchmarks/macrobenchmarks/test_driver/util.dart [hunk 5]
```diff
@@ -0,0 +1,44 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
```
Comment (jonahwilliams) [inline dev/benchmarks/macrobenchmarks/test_driver/util.dart]:
```suggestion
// Copyright 2019 The Chromium Authors. All rights reserved.
```

File: dev/benchmarks/macrobenchmarks/test_driver/cubic_bezier_perf_test.dart [hunk 6]
```diff
@@ -0,0 +1,11 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
```
Comment (jonahwilliams) [inline dev/benchmarks/macrobenchmarks/test_driver/cubic_bezier_perf_test.dart]:
```suggestion
// Copyright 2019 The Chromium Authors. All rights reserved.
```

File: dev/benchmarks/macrobenchmarks/test_driver/cubic_bezier_perf.dart [hunk 7]
```diff
@@ -0,0 +1,40 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
```
Comment (jonahwilliams) [inline dev/benchmarks/macrobenchmarks/test_driver/cubic_bezier_perf.dart]:
```suggestion
// Copyright 2019 The Chromium Authors. All rights reserved.
```

File: dev/benchmarks/macrobenchmarks/lib/src/cubic_bezier.dart [hunk 8]
```diff
@@ -0,0 +1,380 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
```
Comment (jonahwilliams) [inline dev/benchmarks/macrobenchmarks/lib/src/cubic_bezier.dart]:
```suggestion
// Copyright 2019 The Chromium Authors. All rights reserved.
```

Comment (liyuqian) [reply dev/benchmarks/macrobenchmarks/lib/src/cubic_bezier.dart]: Done

File: dev/benchmarks/macrobenchmarks/test_driver/cubic_bezier_perf.dart (hunk 7 above)
Comment (liyuqian) [reply dev/benchmarks/macrobenchmarks/test_driver/cubic_bezier_perf.dart]: Done.

File: dev/benchmarks/macrobenchmarks/test_driver/cubic_bezier_perf_test.dart (hunk 6 above)
Comment (liyuqian) [reply dev/benchmarks/macrobenchmarks/test_driver/cubic_bezier_perf_test.dart]: Done.

File: dev/benchmarks/macrobenchmarks/test_driver/util.dart (hunk 5 above)
Comment (liyuqian) [reply dev/benchmarks/macrobenchmarks/test_driver/util.dart]: Done.

File: dev/devicelab/bin/tasks/cubic_bezier_perf__timeline_summary.dart (hunk 4 above)
Comment (liyuqian) [reply dev/devicelab/bin/tasks/cubic_bezier_perf__timeline_summary.dart]: Done.

File: examples/layers/raw/shader_warm_up.dart (hunk 3 above)
Comment (liyuqian) [reply examples/layers/raw/shader_warm_up.dart]: Done.

File: examples/layers/test/smoketests/raw/shader_warm_up_test.dart (hunk 2 above)
Comment (liyuqian) [reply examples/layers/test/smoketests/raw/shader_warm_up_test.dart]: Done.

**Review by liyuqian — COMMENTED** [review]
Thanks again! Sorry that I forget to check other files...

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Pull request merged**